pennylane/devices/qutrit_mixed/simulate.py

```diff
@@ -0,0 +1,190 @@
+
+    Args:
+        circuit (.QuantumScript): The single circuit to simulate
+        state (TensorLike): The state to perform measurement on
+        is_state_batched (bool): Whether the state has a batch dimension or not.
+        rng (Union[None, int, array_like[int], SeedSequence, BitGenerator, Generator]): A
+            seed-like parameter matching that of ``seed`` for ``numpy.random.default_rng``.
+            If no value is provided, a default RNG will be used.
+        prng_key (Optional[jax.random.PRNGKey]): An optional ``jax.random.PRNGKey``. This is
+            the key to the JAX pseudo random number generator. Only for simulation using JAX.
+            If None, the default ``sample_state`` function and a ``numpy.random.default_rng``
+            will be for sampling.
+
+    Returns:
+        Tuple[TensorLike]: The measurement results
+    """
+
+    circuit = circuit.map_to_standard_wires()
+
+    if not circuit.shots:
+        # analytic case
+
+        if len(circuit.measurements) == 1:
+            return measure(circuit.measurements[0], state, is_state_batched)
+
+        return tuple(measure(mp, state, is_state_batched) for mp in circuit.measurements)
+
+    # finite-shot case
+    rng = default_rng(rng)
+    results = tuple(
+        measure_with_samples(
+            mp,
+            state,
+            shots=circuit.shots,
+            is_state_batched=is_state_batched,
+            rng=rng,
+            prng_key=prng_key,
+        )
+        for mp in circuit.measurements
+    )
+
+    if len(circuit.measurements) == 1:
+        return results[0]
+    if circuit.shots.has_partitioned_shots:
+        return tuple(zip(*results))
+    return results
+
+
+# pylint: disable=too-many-arguments
+def simulate(
+    circuit: qml.tape.QuantumScript,
+    rng=None,
+    prng_key=None,
+    debugger=None,
+    interface=None,
+    # state_cache: Optional[dict] = None, TODO: remove?
+) -> Result:
+    """Simulate a single quantum script.
+
+    This is an internal function that will be called by ``default.qutrit.mixed``.
+
+    Args:
+        circuit (QuantumTape): The single circuit to simulate
+        rng (Union[None, int, array_like[int], SeedSequence, BitGenerator, Generator]): A
+            seed-like parameter matching that of ``seed`` for ``numpy.random.default_rng``.
+            If no value is provided, a default RNG will be used.
+        prng_key (Optional[jax.random.PRNGKey]): An optional ``jax.random.PRNGKey``. This is
+            the key to the JAX pseudo random number generator. If None, a random key will be
+            generated. Only for simulation using JAX.
+        debugger (_Debugger): The debugger to use
+        interface (str): The machine learning interface to create the initial state with
+        # state_cache=None (Optional[dict]): A dictionary mapping the hash of a circuit to the       TODO: remove?
+        #     pre-rotated state. Used to pass the state between forward passes and vjp calculations.
```

Comment:
I don't think this is relevant to the qutrit device. This cache is used for adjoint differentiation, which we don't support with `default.mixed` either.

```suggestion
```

Reply:
I agree, just wanted to check